src/EditorFeatures/Core/Implementation/ReferenceHighlighting/AbstractDocumentHighlightsService.cs

```diff
@@ -53,22 +58,28 @@ internal abstract class AbstractDocumentHighlightsService : IDocumentHighlightsS
             return await SymbolFinder.FindSymbolAtPositionAsync(currentSemanticModel, position, document.Project.Solution.Workspace, cancellationToken).ConfigureAwait(false);
         }
 
-        private async Task<IEnumerable<DocumentHighlights>> GetTagsForReferencedSymbolAsync(
-            ISymbol symbol,
+        private async Task<ImmutableArray<DocumentHighlights>> GetTagsForReferencedSymbolAsync(
+            SymbolAndProjectId symbolAndProjectId,
             IImmutableSet<Document> documentsToSearch,
             Solution solution,
             CancellationToken cancellationToken)
         {
+            var symbol = symbolAndProjectId.Symbol;
             Contract.ThrowIfNull(symbol);
             if (ShouldConsiderSymbol(symbol))
             {
-                var references = await SymbolFinder.FindReferencesAsync(
-                    symbol, solution, progress: null, documents: documentsToSearch, cancellationToken: cancellationToken).ConfigureAwait(false);
-
-                return await FilterAndCreateSpansAsync(references, solution, documentsToSearch, symbol, cancellationToken).ConfigureAwait(false);
+                var progress = new StreamingProgressCollector(
+                    StreamingFindReferencesProgress.Instance);
+                await SymbolFinder.FindReferencesAsync(
+                    symbolAndProjectId, solution, progress,
```

Comment:
this is the bit that makes it so HighlightRefs will now work out of proc.  By passing along a "SymbolAndProjectId" we give enough information to SymbolFinder so it can run out of proc.